### PR TITLE
US131577 - Max Destinations Config Check (Front End)

### PR DESCRIPTION
**Pull request by bamcdole**
[US131577](https://rally1.rallydev.com/#/?detail=/userstory/608879378681&fdp=true)

Back End PR: [https://github.com/Brightspace/lms/pull/14162](https://github.com/Brightspace/lms/pull/14162)

During this implementation, I took into consideration the case where a "Max Destinations" value is not provided to the COCOA BSI component. I elected to do this by using a default value of "0", which is treated as a special case that no longer enforces a max destination count.

In choosing "0" as the special case, it follows that _actually configuring_ the `d2l.Tools.Dropbox.MaxSelectableCoursesCopy` config variable to "0" allows the user to select destinations without a maximum.

How does the team feel about this? The benefit I see is that there would be a already be a config-driven option for disabling the copy limit, which may be desirable in future. The downside is that this is not exactly intuitive, and we may not _want_ to expose this option to customers (though I believe this config is D2LSupport right now).

If we decide _not_ to expose this "no limit mode" via config, I can simply change the default case to "null" instead (which is not allowed by the config variable).

**Demo**
US131577

### CODE REVIEW

**Comment by jorge-ramirez-arredondo** [comment]
Can't configs be made nullable? If we wanted to allow for no limit, we should change the config to be nullable. I think that would make way more sense than setting a max value to 0 meaning there is no max.

Even if we wanted to 0 to mean no limit though, I don't think that should be the logic on the front-end. The front-end should treat null as no limit and then the renderer on the back end should convert a config value of 0 to null (no attribute) when rendering the component.

**Comment by bamcdole** [comment]
> Can't configs be made nullable? If we wanted to allow for no limit, we should change the config to be nullable. I think that would make way more sense than setting a max value to 0 meaning there is no max.
> 
> Even if we wanted to 0 to mean no limit though, I don't think that should be the logic on the front-end. The front-end should treat null as no limit and then the controller on the back end should convert a config value of 0 to null (no attribute) when rendering the component.

Configs can definitely be made nullable, and if we take the time to change it, we should prevent that value from being negative as well 😆

I like your suggestion regarding the backend protecting against a 0 (and I suppose negative) configuration - I'll make the changes.

**Comment by bamcdole** [comment]
Implemented the "null" default instead of "0", per @jorge-ramirez-arredondo's reccomendation.

See the backend PR for the other side of this change.

File: copy-assignment-to-other-courses.js [hunk 1]
```diff
@@ -226,20 +230,24 @@ class CopyAssignmentToOtherCourses extends LocalizeMixin(LitElement) {
...
 						href="${this.enrollmentsRootApi}"
+						maxDestinations=${ifDefined(maxDestinations)}
 						.destinationCourses=${this._selectedCourses}
 						.token=${this.token}>
 					</catoc-course-selection>`;
 			case 1:
 				return html`
 					<catoc-destination-courses
 						class="d2l-destination-courses"
+						maxDestinations=${ifDefined(maxDestinations)}
```
Comment (bamcdole) [inline copy-assignment-to-other-courses.js:250]: For "Number" properties, an omitted attribute results in the property being "null", and a "null" attribute value results in the property being "NaN".

I am getting around these scenarios by turning omitting the sub-components "maxDestination" property if the LMS didn't provide one to the parent component.

**Review by jorge-ramirez-arredondo — APPROVED** [review]

**Pull request merged**